frontend/public/components/replicaset.jsx

```diff
@@ -21,10 +22,36 @@ import {
 
 import { ResourceEventStream } from './events';
 import { VolumesTable } from './volumes-table';
+import { k8sPatch } from '../module/k8s';
+import { confirmModal } from './modals';
 
 const {ModifyCount, AddStorage, common} = Kebab.factory;
 
-export const replicaSetMenuActions = [ModifyCount, AddStorage, ...common];
+const CancelAction = (kind, obj) => ({
+  label: 'Cancel Rollout',
+  hidden: !_.includes(['New', 'Pending', 'Running'], _.get(obj, ['metadata', 'annotations', 'openshift.io/deployment.phase'])),
+  callback: () => confirmModal({
+    title: 'Cancel Rollout',
+    message: 'Are you sure you want to cancel this rollout?',
+    btnText: 'Yes, cancel',
+    cancelText: 'No, don\'t cancel',
+    executeFn: () => k8sPatch(kind,
+      obj,
+      [{
+        op: 'add',
+        path: '/metadata/annotations/openshift.io~1deployment.cancelled',
+        value: 'true',
+      }, {
+        op: 'add',
+        path: '/metadata/annotations/openshift.io~1deployment.status-reason',
+        value: 'cancelled by the user',
+      }]
+    ),
+  }),
+  accessReview: asAccessReview(kind, obj, 'patch'),
+});
+
+export const replicaSetMenuActions = [CancelAction, ModifyCount, AddStorage, ...common];
```

Comment:
You shouldn't add it to replica set menu actions. It is only applicable to replication controllers. Move the action to replication-controller.jsx and prepend it to the array there.